viral/ensemble_reactivation.py

```
@@ -0,0 +1,616 @@
+    activity estimates were calculated across time using the fastICA algorithm. Only
+    the subset of components found to be significant under the Marcenko-Pasteur
+    distribution were considered ICA ensembles and included in the ICA ensemble
+    matrix w, with rows corresponding to PCs and columns corresponding to
+    significant ICA components."""
+    # TODO: keep random state as ICA is non-deterministic to keep results reproducible?
+    ica_transfomer = FastICA(random_state=999)
+    # expected input: (n_samples, n_features)
+    # Thijs van der Plas: features = place cells, samples = frames, components = ICA components
+    # ssp vectors (place cells, frames), ssp vectors.T (frames, place cells)
+    ica_components = ica_transfomer.fit_transform(ssp_vectors.T)
+    significant_components = detect_significant_components(ica_components)
```

Pylint (3.3.7)

[error] No value for argument 'eigenvalues' in function call

(E1120)